### PR TITLE
Dv routing

### PR DESCRIPTION
**Pull request by Kiricon**
Changes:
1) Added `match` method for general routing.
2) Added 404 page and routing (not styled yet)
3) Added 500 error page with in depth stack tracing and and integrated editor (File is large, open with sublime, inlining the ace editor really messed it up)
4) Renaming routing files, but honestly I don't know how to make the names much better :(

I will continue working in this branch and making the changes we had last spoken of for the router.

### CODE REVIEW

File: core.js [hunk 1]
```diff
@@ -31,7 +32,12 @@ module.exports.startServer = function() {
 
     // Register all our routes with express
     let Routes = [];
-    Routes.push(require(currentPath + "/" + Config.routesFolder + "/routes.js"));
+    Routes.push(
+        require(currentPath + '/' + Config.routesFolder + '/routes.js')
```
Comment (ava-cassiopeia) [inline core.js]: Is there a particular reason why we're calling it `Config` instead of `config`?

File: core/route/allrouting.js [hunk 2]
```diff
@@ -87,6 +87,18 @@ class Routing {
...
 
     /**
+     * Match Method
+     * @param {String[]} methods - Array of methods to use for a route
+     * @param {String} path - Path for routing
+     * @param {String|Function} controllerAction
+     */
+    match(methods, path, controllerAction) {
+        methods.forEach((method) => {
+            this[method.toLowerCase()](path, controllerAction);
```
Comment (ava-cassiopeia) [inline core/route/allrouting.js]: I would maybe have a constant array in this class something like

```
const allowedMethod = ["get", "post", "patch", "put"];
```

And run `method.toLowerCase()` through that first, just so people know when they typed something wrong.

**Review by ava-cassiopeia — CHANGES_REQUESTED** [review]
See my comments. Totally open to discussion on those. Otherwise code looks good.

File: core.js (hunk 1 above)
Comment (Kiricon) [reply core.js]: Nope, I'll just rename it real quick. It is inconsistent.

Comment (ava-cassiopeia) [reply core.js]: Awesome, thanks.

File: core/route/allrouting.js (hunk 2 above)
Comment (Kiricon) [reply core/route/allrouting.js]: I had thought about that but had overlooked it at the time due to us not having a consistent means for returning errors. Regardless we will need it in the end so I'll add that fix real quick.

Comment (ava-cassiopeia) [reply core/route/allrouting.js]: Yeah, true. For now I'd just say make a class that extends `Error` like `IllegalMethodError` and just throw that. We can do more with catching that later.

Comment (Kiricon) [reply core/route/allrouting.js]: I was planning on making error handling a full task for the future. Good errors handling is a big help.

Comment (ava-cassiopeia) [reply core/route/allrouting.js]: Agreed 100%.

**Review by ava-cassiopeia — APPROVED** [review]
👍

**Pull request merged**